src/pages/index.js

```diff
@@ -0,0 +1,311 @@
+const previewImageModal = document.querySelector("#preview__modal");
+const previewImageModalCloseButton =
+  previewImageModal.querySelector(".modal__close");
+const previewImageModalOverlay = document.querySelector("#preview__modal");
+
+//Delete Modal
+const deleteModal = document.querySelector("#delete__modal");
+const deleteModalButton = deleteModal.querySelector(".modal__button");
+const deleteModalCloseButton = deleteModal.querySelector(".modal__close");
+
+//Avatar Modal
+const avatarModal = document.querySelector("#avatar__modal");
+const avatarEditButton = document.querySelector(".profile__avatar-edit");
+const avatarModalCloseButton = avatarModal.querySelector(".modal__close");
+const avatarModalSubmitButton = avatarModal.querySelector(".modal__button");
+
+//////////////////////////////////////////////////// CLASSES  /////////////////////////////////////////////////////////////////
+//Instantiate Section Class
+const section = new Section(
+  { item: initialCards, renderer: renderCard },
+  ".cards__list"
+);
+////Instantiate UserInfo
+const user = new UserInfo({
+  name: ".profile__title",
+  job: ".profile__description",
+  avatar: ".profile__image",
+});
+////
+//Instantiate popupWithForm for Profile
+const profilePopup = new PopupWithForm(
+  "#profile__edit-modal",
+  handleProfileModalSubmit
+);
+profilePopup.setEventListeners();
+
+//Instantiate popupWithForm for addCard
+const newCardPopup = new PopupWithForm(
+  "#card__add-modal",
+  handleAddCardFormSubmit
+);
+newCardPopup.setEventListeners();
+//////Preview Image popup class
+const popupImage = new PopupWithImage("#preview__modal");
+popupImage.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#avatar__modal", handleAvatarUpdate);
+avatarPopup.setEventListeners();
+
+const deleteConfirm = new PopupWithConfirm("#delete__modal");
+deleteConfirm.setEventListeners();
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4792ec92-cf1c-45a6-8740-0f5d63585faa",
+    "Content-type": "application/json",
+  },
+});
+
+//////////////////////////////////////////// FUNCTIONS //////////////////////////////////////////////////////
+
+function handleImageClick(name, link) {
+  popupImage.openPopup({ name, link });
+}
+
+function handleProfileModalSubmit(profileData) {
+  profileModalSubmitButton.textContent = "Saving...";
+  console.log("profileData:", profileData);
+  const name = profileData.title; //set name var to work with userInfo class
+  const job = profileData.description; //set job var to work with userInfo class
+  console.log("name:", name, "job:", job); //debug log
+  user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
+  const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
+  console.log(updatedUserInfo); // debug log
+  profilePopup.closePopup(); //close popup
+
+  setTimeout(() => {
```

Comment:
`setTimeout` is not needed in your project because you need to wait naturally for the server response. You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.